src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -138,16 +138,29 @@ sealed class NumericRange[T](
   // based on the given value.
   private def newEmptyRange(value: T) = NumericRange(value, value, step)
 
+  // Calling length when a range contains more than Int.MaxValue causes IllegalArgumentException
+  // because the length of the range cannot be represented with Int numbers.
+  // This method tries to compare the length of this with the given Int number.
+  private def compareLength(n: Int): Int = {
+    val diff = num.minus(end, start)
+    val quotient = num.quot(diff, step)
+    val remainder = num.minus(diff, num.times(quotient, step))
+    val thisLength = if (!isInclusive && zero == remainder) quotient else num.plus(quotient, num.one)
```

Comment:
This looks orders of magnitues simpler than [`NumericRange.count`](https://github.com/scala/scala/blob/v2.13.10/src/library/scala/collection/immutable/NumericRange.scala#L292) - did you take a look at this code? Is it unnecessarily complicated?

Reply:
I also think that `NumericRange.count` is over-complicated. As far as I can think of now, we should be able to use this as `NumericRange.count` with some minor modifications, like adding this if condition at the end of the method so that it can fit in Integers:
```scala
if (num.gt(thisLength, Int.MaxValue))
  throw new IllegalArgumentException("...") 
else num.toInt(thisLength)
```
I can also try this locally and run the tests to check if it succeeds.

Reply:
I was just looking at my old branch for issue 12073, and also the couple of fixes since the rewrite. I'd be surprised if the complexity were spurious, but it will take me some time to reabsorb anything. Of course, I am often surprised and wrong. Odersky has commented that `NumericRange` is an overengineered overabstraction. Maybe the overs go together.

Reply:
I started implementing a simpler implementation of `NumericRange.count` locally, and as I run the tests, I encountered some issues with `BigDecimal`s, although it worked fine for other `Integral`s. So I guess it can be simplified well for other types, but it'd still need some exceptional parts for `BigDecimal`.

Reply:
The comments in `NumericRange.count` explain why it's as complex as it is.